app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,45 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      providers = {
+        :gce => [ManageIQ::Providers::Google::CloudManager, :name],
+        :aws => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      instance_uri = URI(provider_id)
```

Comment:
@enoodle using `URI` here seems an overkill (also because there's no contract for the providerID to respect the URI format). This would be enough:

```
scheme, instance_uri = provider_id.to_s.split("://", 2)
```

Make sure that this is not failing for special values of `provider_id`.